lms/services/canvas_api/public.py

```diff
@@ -0,0 +1,236 @@
+"""High level access to Canvas API methods."""
+
+import marshmallow
+from marshmallow import EXCLUDE, Schema, fields, post_load, validate, validates_schema
+
+from lms.services import CanvasAPIError
+from lms.validation import RequestsResponseSchema
+
+
+class _SectionSchema(Schema):
+    """
+    Schema for an individual course section dict.
+
+    These course section dicts appear in various different Canvas API responses.
+    This _SectionSchema class is used as a base class or nested schema by
+    various schemas below for Canvas API responses that contain section dicts.
+    """
+
+    class Meta:
+        unknown = EXCLUDE
+
+    id = fields.Int(required=True)
+    name = fields.String(required=True)
+
+
+class CanvasAPIClient:
+    """
+    A client for making calls to the CanvasAPI.
+
+    All methods may raise:
+
+    :raise CanvasAPIAccessTokenError: if we can't get the list of sections
+            because we don't have a working Canvas API access token for the user
+    :raise CanvasAPIServerError: if we do have an access token but the
+            Canvas API request fails for any other reason
+    """
+
+    def __init__(self, authenticated_client):
+        """
+        Create a new CanvasAPIClient.
+
+        :param authenticated_client: An instance of AuthenticatedClient.
+        """
+        self._api = authenticated_client
+
+    def get_token(self, authorization_code):
+        """
+        Get an access token for the current LTI user.
+
+        :param authorization_code: The Canvas API OAuth 2.0 authorization code
+                                   to exchange for an access token
+        :return: An access token string
+        """
+        return self._api.get_token(authorization_code)
+
+    def authenticated_users_sections(self, course_id):
+        """
+        Return the authenticated user's sections for the given course_id.
+
+        :param course_id: the Canvas course_id of the course to look in
+        :return: a list of raw section dicts as received from the Canvas API
+        :rtype: list(dict)
+        """
+
+        return self._ensure_sections_unique(
```

Comment:
I think this looks like it should be a decorator (`@unique_sections` or something) we can drop on this function. This however is very much a job for another day.